src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -256,142 +368,173 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
         .withName("ARMADA_SPARK_DRIVER_SERVICE_NAME")
-        .withValue(driverServiceName)
+        .withValue(serviceName)
     )
-
-    val configGenerator =
-      new ConfigGenerator("armada-spark-config", conf)
-    val primaryResource = clientArguments.mainAppResource match {
-      case JavaMainAppResource(Some(resource)) => Seq(resource)
-      case PythonMainAppResource(resource)     => Seq(resource)
-      case RMainAppResource(resource)          => Seq(resource)
-      case _                                   => Seq()
-    }
-
-    val confSeq = conf.getAll.flatMap { case (k, v) =>
-      Seq("--conf", s"$k=$v")
-    }
-    val sparkDriverPort = 7078
-    val driverContainer = Container()
-      .withName("spark-driver")
+    Container()
+      .withName("driver")
       .withImagePullPolicy("IfNotPresent")
-      .withImage(conf.get("spark.kubernetes.container.image"))
+      .withImage(image)
       .withEnv(envVars)
       .withCommand(Seq("/opt/entrypoint.sh"))
-      .withVolumeMounts(configGenerator.getVolumeMounts)
+      .withVolumeMounts(volumeMounts)
       .withPorts(
         Seq(
-          ContainerPort(
-            Option("as-driver-port"),
-            Option(0),
-            Option(sparkDriverPort)
-          )
+          ContainerPort(name = Option("driver"), containerPort = Option(port))
         )
       )
       .withArgs(
         Seq(
           "driver",
           "--verbose",
-          "--class",
-          clientArguments.mainClass,
           "--master",
-          "local://armada://armada-server.armada.svc.cluster.local:50051",
+          master,
+          "--class",
+          mainClass,
           "--conf",
-          s"spark.driver.port=$sparkDriverPort",
+          s"spark.driver.port=$port",
           "--conf",
           "spark.driver.host=$(SPARK_DRIVER_BIND_ADDRESS)"
-        ) ++ confSeq ++ primaryResource ++ clientArguments.driverArgs
+        ) ++ additionalDriverArgs
       )
       .withResources( // FIXME: What are reasonable requests/limits for spark drivers?
         ResourceRequirements(
-          limits = Map(
-            "memory"            -> Quantity(Option("450Mi")),
-            "ephemeral-storage" -> Quantity(Option("512Mi")),
-            "cpu"               -> Quantity(Option("200m"))
-          ),
-          requests = Map(
-            "memory"            -> Quantity(Option("450Mi")),
-            "ephemeral-storage" -> Quantity(Option("512Mi")),
-            "cpu"               -> Quantity(Option("200m"))
-          )
+          requests = defaultDriverResources,
+          limits = defaultDriverResources
         )
       )
+  }
 
-    val executorContainers =
-      getExecutorContainers(numExecutors, driverServiceName, conf)
-
-    val gangAnnotations = GangSchedulingAnnotations(
-      None,
-      1 + numExecutors,
-      conf.get(ARMADA_JOB_GANG_SCHEDULING_NODE_UNIFORMITY)
+  private val defaultDriverResources = Map(
+    "memory" -> Quantity(Option("512Mi")),
```

Comment:
these should be configurable

Reply:
I'll do it in a followup PR, it is already very big.